festim/geometry/mesh_1d.py

```diff
@@ -0,0 +1,72 @@
+from dolfinx import fem, mesh
+from mpi4py import MPI
+import ufl
+import numpy as np
+from festim import Mesh
+
+
+class Mesh1D(Mesh):
+    """
+    1D Mesh
+    Attributes:
+        vertices (list): the mesh x-coordinates
+        size (float): the size of the 1D mesh
+        V (dolfinx.fem.FunctionSpace): the function space of the simulation
+    """
+
+    def __init__(self, vertices, **kwargs) -> None:
+        """Inits Mesh1D
+
+        Args:
+            vertices (list): the mesh x-coordinates
+        """
+
+        self.vertices = vertices
+
+        self.start = min(vertices)
+        self.size = max(vertices)
+        self.V = None
+
+        mesh = self.generate_mesh()
+        super().__init__(mesh=mesh, **kwargs)
+
+    def generate_mesh(self):
+        """Generates a 1D mesh"""
+        gdim, shape, degree = 1, "interval", 1
+        cell = ufl.Cell(shape, geometric_dimension=gdim)
+        domain = ufl.Mesh(ufl.VectorElement("Lagrange", cell, degree))
+
+        mesh_points = np.reshape(self.vertices, (len(self.vertices), 1))
+        indexes = np.arange(self.vertices.shape[0])
+        cells = np.stack((indexes[:-1], indexes[1:]), axis=-1)
+
+        return mesh.create_mesh(MPI.COMM_WORLD, cells, mesh_points, domain)
+
+    def define_surface_markers(self, function_space):
```

Comment:
I understand now the function_space thing.
So let me rephrase what I said earlier:

One mesh and a set of subdomains can be applicable to several different models.
HOWEVER, the mesh tags are only relevant for a given simulation since they are linked to the function space (and because the function space is unique for the simulation)

I think we therefore need to change the way we do things. 

We can keep this method cause it's still Geometry related, but just not store non-geometry-unique attributes in Geometry (so mesthtags, measures)